server/chat-plugins/battlesearch.ts

```diff
@@ -0,0 +1,410 @@
+/**
+ * Battle search - handles searching battle logs.
+ */
+import * as child_process from 'child_process';
+import * as util from 'util';
+
+import {FS} from '../../lib/fs';
+import {Utils} from '../../lib/utils';
+import {QueryProcessManager} from '../../lib/process-manager';
+import {Repl} from '../../lib/repl';
+import {checkRipgrepAvailability} from '../config-loader';
```

Comment:
This is not what I meant by "put it in config loader" but I guess it's fine for now; no need to stall this PR any longer.